### DISCUSSION TITLE
Using sdmTMB to model presence-only fisheries data

### DISCUSSION THREAD

**Discussion by ericward-noaa** [sdmTMB/sdmTMB#312] [General]
From Issues             _Originally posted by @nhill917 in https://github.com/pbs-assess/sdmTMB/issues/84#issuecomment-1977623705_
    
Hi there, I would like to use sdmTMB to model the distribution of a number of species using
presence-only fisheries observer data. I am interested in determining species ranges and their probability 
of occurrence, similar to common methods like Maxent, rather than abundance. I am interested in using 
sdmTMB so that I can attempt to remove/estimate the effect of preferential sampling/non-independence 
of fishing from the data. What is the best way to structure the model in sdmTMB to achieve this? Would 
it be best to use a similar approach as that used in cpue standardisation where I include effort, fleet etc 
as covariates along with environmental covars like sst in a negative binomial, presence-absence model 
and attempt to standardise them out? Or is it possible/preferable to use an approach like that in Pennino 
et al 2018 or Conn et al 2017 (Accounting for preferential sampling in species distribution models; 
Confronting preferential sampling when analysing population distributions: diagnosis and model-based 
triage), where it seems (not clear to me) a weighting surface is produced and a 2 step model used? If 
so, how can I achieve this within sdmTMB? Any help is much appreciated, this is a great resource. 
Kind regards.

**Comment by ericward-noaa** [comment]
@nhill917 Good question, I think we could update the vignette to include an example along these lines. 

The approaches from the Conn or Pennino papers should be do-able in sdmTMB. I think you're going to have to do this in several steps regardless, because you'll need to construct the pseudo-absences and do a sensitivity to that choice. I think a general strategy is to generate them over a range of covariates / pressures (including fishing effort) - so this will probably be different than the spatially uniform approach taken in the vignette, https://github.com/pbs-assess/sdmTMB-teaching/blob/main/noaa-psaw-2022/06-presence-only.Rmd

Downweighted Poisson Regression is a good choice for estimation. See how the Conn et al. paper constructed their B matrix -- you should be able to do something similar and test for the PS effect

**Comment by nhill917** [comment]
Thanks for the quick response Eric, a vignette or some help would be much appreciated. My plan was to build raster layers of effort metrics (# hooks etc) from the total observer dataset so that I can extract data to the background points the same as the presence points. I guess I am stuck as to how to formulate the two step process. ie: what the sdmTMB() function will look like for each of the two models and how information from the first is fed into the second. Kind regards.

**Comment by seananderson** [comment]
That Rmd Eric [linked to](https://github.com/pbs-assess/sdmTMB-teaching/blob/main/noaa-psaw-2022/06-presence-only.Rmd) should have a complete example. The output from that source code is here: https://pbs-assess.github.io/sdmTMB-teaching/noaa-psaw-2022/06-presence-only.html#1

There is a draft vignette based on that set of slides, that it's long been on my todo list to integrate as an official vignette:
https://github.com/pbs-assess/sdmTMB/blob/main/scratch/julia-vignettes/presence-only.Rmd

**Comment by nhill917** [comment]
Thank you sean. The vignettes and slides provided are great, and I even found the associated youtube clip going through the presentation. What I am struggling with is how to extend these approaches to incorporate biased/preferential sampling similar to the approaches discussed in the papers in the initial comment. I am using fisheries observer data to explore the distribution of various bycatch species, so the data is very patchy/biased. It'd be great to use sdmTMB to understand the distribution of these species, but also how the preferential sampling influences the model and likely how other, more conventional approaches fail to account for these influences.

**Reply by seananderson** [reply]
The short answer is, no, explicitly dealing with preferential sampling isn't currently possible. However, I'm very interested in incorporating this and have started experimenting with some options locally. The classic option would be linking a random field associated with a point process defining sampling locations (likely fit as defined above) into the linear predictor of the density (with some associated scaling coefficient on that linked random field). An alternative that I haven't yet fully wrapped my head around would be something like in [this](https://onlinelibrary.wiley.com/doi/abs/10.1002/eap.2453) paper. Fitting the common kind of data you're talking about is a bit more complex than in [this](https://onlinelibrary.wiley.com/doi/abs/10.1111/2041-210X.12803) paper since we usually don't have a clear set of stations that we chose not to sample.
